cli/cmd/root.go

```diff
@@ -130,7 +131,9 @@ func runVersionCheck(vChecker *version.VersionChecker, flags []string, cliConfig
 		}
 
 		if clusterVersion != Version {
-			fmt.Fprintf(os.Stderr, "* Warning: Your Keptn CLI version (%s) and Keptn cluster version (%s) don't match. This can lead to problems. Please make sure to use the same versions.\n", Version, clusterVersion)
+			if len(flags) == 0 || flags[0] != "upgrade" {
```

Comment:
Hi, thank you for your contribution! There is actually no need to test the length of the `flags` to 0. According to the implementation, this can never happen in this piece of code. I suggest this:

`if flags[0] != "upgrade"`

I understand that afterwards you will receive a SIGSEGV error during running the unit tests, but that's only due to missing `args` parameter in the input struct, adding this parameter, for example like `args: []string{"status"},` should solve the problem on tests side